host-controller/src/main/java/org/jboss/as/domain/controller/operations/deployment/DeploymentExplodeHandler.java

```diff
@@ -115,6 +115,7 @@ public void execute(OperationContext context, ModelNode operation) throws Operat
             newHash = DeploymentUploadUtil.synchronizeSlaveHostController(operation, address, fileRepository, contentRepository, backup, oldHash);
         }
         if (newHash != null) {
+            contentRepository.addContentReference(ModelContentReference.fromModelAddress(context.getCurrentAddress(), newHash));
```

Comment:
To be picky, there is no ref to this content until a few lines later.

Also, the ResultHandler is not added unless contentRepository != null, which means the add/remove may not be consistent. I think actually the contentRepo is never null. If that's true it would be good to clean the null checking up and simplify this code.